src/components/common/FileUploadPdf.vue

```diff
@@ -0,0 +1,295 @@
+    id="file-upload-pdf"
+    dense
+    filled
+    show-size
+    color="primary"
+    hide-details="auto"
+    accept=".pdf"
+    label="Select a file to upload"
+    :value="file"
+    :error-messages="errorMessages"
+    @change="onChange($event)"
+  />
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Emit, Prop } from 'vue-property-decorator'
+import { GlobalWorkerOptions, getDocument } from 'pdfjs-dist/build/pdf'
+import { PageSizes } from '@/enums'
+import { PdfInfoIF, PresignedUrlIF } from '@/interfaces'
+import { AxiosResponse } from 'axios'
+
+// set web worker
+GlobalWorkerOptions.workerSrc = require('pdfjs-dist/build/pdf.worker.entry')
+
+@Component({})
+export default class FileUploadPdf extends Vue {
+  @Prop({ default: null }) readonly file!: File
+  @Prop({ default: true }) readonly isRequired!: boolean
+  @Prop({ default: 0 }) readonly maxSize!: number // in MB
+  @Prop({ default: null }) readonly pageSize!: PageSizes
+  @Prop({ required: true }) readonly userId!: string
+
+  // Network service functions
+  @Prop({ required: true })
+  readonly getPresignedUrl!: (fileName: string) => Promise<PresignedUrlIF>
+  @Prop({ required: true })
+  readonly uploadToUrl!: (url: string, file: File, key: string, userId: string) => Promise<AxiosResponse>
+
+  /** Dictionay of page size(s). */
+  readonly PAGE_SIZE_DICT = {
+    LETTER_PORTRAIT: {
+      pointsPerInch: 72,
+      width: 8.5,
+      height: 11,
+      validationErrorMsg: 'Document must be set to fit onto 8.5” x 11” letter-size paper'
+    }
+  }
+
+  /** Upload failed message. */
+  readonly UPLOAD_FAILED_MESSAGE = 'An error occurred while uploading. Please try again.'
+
+  // Local properties
+  protected fileKey = null as string
+  protected errorMessages = [] as Array<string> // custom error messages
+  protected validFile = false
+
+  /**
+   * Clears input and resets validation errors.
+   * FUTURE: delete file from Minio
+   */
+  public reset (): void {
+    this.errorMessages = []
+    this.emitUpdateFile(null)
+    this.emitFileKey(null)
+  }
+
+  /** Resets input validation only. */
+  public resetValidation (): void {
+    this.errorMessages = []
+  }
+
+  /** Validates this component. */
+  public validate (): boolean {
+    // check that file exists if required
+    if (this.isRequired && !this.file) {
+      this.errorMessages = ['File is required']
+      return false
+    }
+    // return whether file is valid
+    return this.validFile
+  }
+
+  /** Called when user has changed the input file. */
+  protected async onChange (file: File): Promise<void> {
+    console.log('*** onChange(), file =', file)
+
+    // reset before validation
+    this.fileKey = null
+    this.errorMessages = []
+
+    // TODO:
+    // 1. validate file
+    // 2. upload file to Minio
+    // 3. emit file
+    // 4. emit file key
+
+    // *** TODO: need to check if required file exists???
+
+    if (file) {
+      this.errorMessages = ['Processing...']
```

Comment:
New: show this while validating and uploading the PDF, which can take a few seconds.